src/java.base/share/classes/java/security/Provider.java

```diff
@@ -1896,7 +1906,7 @@ public Object newInstance(Object constructorParameter)
                         null : constructorParameter.getClass();
                 } else {
                     ctrParamClz = cap.constructorParameterClassName == null?
-                        null : Class.forName(cap.constructorParameterClassName);
+                        null : cap.getConstructorParameterClass();
```

Comment:
Maybe we should move the ternary for `cap.constructorParameterClassName == null` into new method to begin with.

Reply:
Thanks. Similar reasoning as above. If we decide to diverge from the original implementation then I'll update this too.